### PR TITLE
 feat(structured date): unique table names for SQLite databases

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

first step for https://github.com/dust-tt/tasks/issues/637

We are currently enforcing uniqueness on the table name per data source. The rationale is that we need unique table names when constructing the database or returning the schema, as a SQLite database cannot have several tables with the same name (obviously) and a model cannot generate a proper query if several tables have identical names. This is problematic because:
- it forces us to add some unique identifier in table names (from eg google drive) to make them unique, often leading to very long, convoluted, and meaningless table names, which negatively impacts model performance.
- even adding those identifiers, we sometimes run into errors
- the unique identifiers make it really hard for users to understand the assistant builder UI, as they see some weird names instead of the actual name of their spreadsheets / databases / csv.

This PR removes the constraint at database level, and adds a deterministic algo to compute unique table names from a set of tables right before returning a DB schema and right before constructing the in-memory sqlite DB.

Next step is to update connectors code to remove the unique identifiers from table names

## Risk

blast radius limited to tables query

## Deploy Plan

deploy code, run migration

### CODE REVIEW

File: core/src/databases/database.rs [hunk 1]
```diff
@@ -448,6 +454,28 @@ impl HasValue for QueryResult {
...
+        .map(|table| {
+            let base_name = table.name();
+            let count = name_count.entry(base_name).or_insert(0);
+            *count += 1;
+
+            (
+                table.unique_id(),
+                match *count {
+                    1 => base_name.to_string(),
+                    _ => format!("{}_{}", base_name, count),
```
Comment (spolu) [inline core/src/databases/database.rs:472]: maybe `count-1` here?

**Review by spolu — APPROVED** [review]
LGTM maybe use `count-1` so that we have `base_name` and `base_name_1` instead of `base_name_2` directly?

File: core/src/databases/database.rs (hunk 1 above)
Comment (fontanierh) [reply core/src/databases/database.rs:472]: Fair

**Review by flvndvd — APPROVED** [review]
LGTM, just wondering how does this work in our caching system?

**Comment by fontanierh** [comment]
@flvndvd 

> LGTM, just wondering how does this work in our caching system?

No impact. Caching is per DB and cache key is based on table unique IDs

**Pull request merged**